device/api/umd/device/pci_device.hpp

```diff
@@ -219,12 +154,24 @@ class PCIDevice {
 
     uint32_t read_checking_offset;
 
-private:
-    bool is_hardware_hung();
-
     template <typename T>
-    T *get_register_address(uint32_t register_offset);
+    T *get_register_address(uint32_t register_offset) {
```

Comment:
why was implementation moved to .hpp file?

Reply:
This is a mess and we might refactor it anyway.
But this is a template function, it has to be implemented in header file. We got away with it before because it was private, and thus only used in pci_device.cpp